src/BlockTypes/MiniCart.php

```diff
@@ -219,6 +226,34 @@ protected function get_markup() {
 			$cart_contents_total += $cart->get_subtotal_tax();
 		}
 
+		$wrapper_classes = 'wc-block-mini-cart wp-block-woocommerce-mini-cart';
+		$classes         = '';
+		$style           = '';
```

Comment:
I didn't test, so no idea if it would work, but just asking. Did you try attaching the styles in an inline `<style>` tag? This way we wouldn't need to read the styles again from the JS, we would just let the selectors work.

Reply:
This is needed for inline style attributes instead. When the users choose a preset color, CSS selectors work fine, but if users choose custom colors, we inline style for them.

Initially, I only add inline style and classes for background and inherit the color from the parent element, but with the change to the global style, I do the same for color as well.

Reply:
To be more clear, previously, before adding experimental selectors, we add classes and style to the wrapper element. As mentioned above, it doesn't work for the background.

After adding experimental selectors, we can target the CSS selectors to apply the global style. But it has a limitation: we can't target a specific selector for a specific style rule. We apply background style for the button and the badge (or the counter bubble, in the other words), the text color will come with it as well:

```
.wc-block-mini-cart__button, .wc-block-mini-cart__badge {
	background-color: var(--wp--preset--color--pale-pink);
	color: var(--wp--preset--color--vivid-purple);
}
```
Because of that, I decided to apply CSS classes and inline style to both button and badge.